File: app-miyou/src/main/java/com/miyou/app/domain/monitoring/model/StagePerformanceRollup.java
package com.miyou.app.domain.monitoring.model;

import java.time.Instant;

public record StagePerformanceRollup(
	Instant bucketStart,
	com.miyou.app.domain.monitoring.model.MetricsGranularity granularity,
	String stageName,
	long count,
	long totalDurationMillis,
	double avgDurationMillis
) {
}

File: domain/src/main/kotlin/com/miyou/app/domain/monitoring/model/StagePerformanceRollup.kt
package com.miyou.app.domain.monitoring.model

import java.time.Instant

data class StagePerformanceRollup(
    val bucketStart: Instant,
    val granularity: MetricsGranularity,
    val stageName: String,
    val count: Long,
    val totalDurationMillis: Long,
    val avgDurationMillis: Double,
)
